File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/template/HeaderTemplateViewHolder.java
package com.supercilex.robotscouter.ui.scout.viewholder.template;

import android.view.View;

import com.supercilex.robotscouter.ui.scout.viewholder.HeaderViewHolder;

public class HeaderTemplateViewHolder extends HeaderViewHolder implements ScoutTemplateViewHolder {
    public HeaderTemplateViewHolder(View itemView) {
        super(itemView);
    }

    @Override
    protected void bind() {
        super.bind();
        mName.setOnFocusChangeListener(this);
    }

    @Override
    public void requestFocus() {
        mName.requestFocus();
    }

    @Override
    public void onFocusChange(View v, boolean hasFocus) {
        if (!hasFocus) updateMetricName(mName.getText().toString());
    }
}

File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/template/HeaderTemplateViewHolder.kt
package com.supercilex.robotscouter.ui.scout.viewholder.template

import android.view.View

import com.supercilex.robotscouter.ui.scout.viewholder.HeaderViewHolder

class HeaderTemplateViewHolder(itemView: View) : HeaderViewHolder(itemView), ScoutTemplateViewHolder {
    override fun bind() {
        super.bind()
        name.onFocusChangeListener = this
    }

    override fun requestFocus() {
        name.requestFocus()
    }

    override fun onFocusChange(v: View, hasFocus: Boolean) {
        if (!hasFocus) updateMetricName(name.text.toString())
    }
}
